lib/smack/SmackRep.cpp

```diff
@@ -239,16 +340,16 @@ const Expr* SmackRep::lit(const llvm::Value* v) {
       Expr::lit(exponentPart));
 
   } else if (llvm::isa<llvm::ConstantPointerNull>(v))
-    return Expr::lit(0, width);
+    return Expr::id("$NULL");
 
   else
     return expr(v);
   // assert( false && "value type not supported" );
 }
 
-const Expr* SmackRep::lit(unsigned v) {
-  // TODO why doesn't this one do the thing with negative as well?
-  return Expr::lit(v, width);
+// Shaobo: if we add multiple types to SMACK, then integer literals generated by SMACK source code should have the sense of type widths
+const Expr* SmackRep::lit(int v, unsigned size) {
```

Comment:
If an integer is used as `size_t`, then we could translate it via `lit(i, SIZE)`, where `SIZE` is a class member holding the value of `size_t`. A program can possibly contain integer constants of which type ranging from `i8` to `i64` and they can be easily handled by this method. So I was wondering if we could keep this method.